### PR TITLE
LPS-137740 Get Application ID by cookie or session storage

### PR DESCRIPTION
**Pull request by RodrigoCavalcantii**

### CODE REVIEW

**Review by kevenleone — CHANGES_REQUESTED** [review]
About the implementation, this action to get a cookie even being very particular, and used one single time, but could be used a function for it.

Something like:

`getCookie(name) {}`

**Comment by kevenleone** [comment]
Hey @RodrigoCavalcantii , I found some implementation that could help us with that:

image

From: https://daily-dev-tips.com/posts/vanilla-javascript-cookies-%F0%9F%8D%AA/

**Comment by kevenleone** [comment]
And this other one:

image

It's looks like better

From: https://gomakethings.com/how-to-set-and-get-cookies-with-vanilla-js/

**Comment by kevenleone** [comment]
@RodrigoCavalcantii , you did a good job on that, this is just a suggestion for improvement.
Because is simpler to understand the code scoped in a function

**Comment by RodrigoCavalcantii** [comment]
> @RodrigoCavalcantii , you did a good job on that, this is just a suggestion for improvement.
> Because is simpler to understand the code scoped in a function

Yes sure, I agree. I’m gonna do it now !

**Comment by kevenleone** [comment]
Just starting reviewing :)

File: modules/apps/site/site-insurance-site-initializer/src/main/zippableResources/fragments/raylife/fragments/get-in-touch/index.js [hunk 1]
```diff
@@ -15,13 +11,19 @@ btnCall.onclick = function(){
...
-	while (auxCookie.charAt(0) == ' ') {
-		auxCookie = auxCookie.substring(1);
+function getCookie(name) {
+	name = name + '=';
+	const decodedCookie = decodeURIComponent(document.cookie);
+	const cookies = decodedCookie.split(';');
+	for (let i = 0; i < cookies.length; i++) {
+		const cookie = cookies[i].trim();
+		if (cookie.indexOf(name) == 0) {
+		return cookie.substring(name.length, cookie.length);
```
Comment (kevenleone) [inline modules/apps/site/site-insurance-site-initializer/src/main/zippableResources/fragments/raylife/fragments/get-in-touch/index.js:21]: Wrong spacing, tab is missing here

File: modules/apps/site/site-insurance-site-initializer/src/main/zippableResources/fragments/raylife/fragments/get-in-touch/index.js [hunk 2]
```diff
@@ -15,13 +11,19 @@ btnCall.onclick = function(){
...
 	}
-	if (auxCookie.indexOf(nameCookie) == 0) {
-		applicationId = auxCookie.substring(nameCookie.length, auxCookie.length);
-		document.getElementById("content-agent-text-your-application").textContent = "Your Application #"+applicationId;
-	}
-}
\ No newline at end of file
+}
+
+if(getCookie(nameCookie) != undefined && getCookie(nameCookie) != "" )
```
Comment (kevenleone) [inline modules/apps/site/site-insurance-site-initializer/src/main/zippableResources/fragments/raylife/fragments/get-in-touch/index.js:26]: this line could be replaced just for `if (getCookie(nameCookie)) {}`

Any string inside this variable or function return, is already true

Comment (kevenleone) [reply modules/apps/site/site-insurance-site-initializer/src/main/zippableResources/fragments/raylife/fragments/get-in-touch/index.js:26]: Also, you are using `getCookie` 3 times, you could assign to one variable, and use it

**Review by kevenleone — CHANGES_REQUESTED** [review]
Hey @RodrigoCavalcantii , I did some observations to the code, it's good, just some improvements to do

**Comment by kevenleone** [comment]
Resent: #167

**Pull request closed without merging**